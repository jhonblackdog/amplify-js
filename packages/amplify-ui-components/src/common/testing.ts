export const pixelThreshold = 10;
