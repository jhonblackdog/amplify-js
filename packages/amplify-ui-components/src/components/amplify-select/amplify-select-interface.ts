interface SelectOption<T> {
  label: string;
  value: T;
}

interface SelectOptions<T extends string | number> extends Array<SelectOption<T>> {}

export type SelectOptionsString = SelectOptions<string>;
export type SelectOptionsNumber = SelectOptions<number>;
