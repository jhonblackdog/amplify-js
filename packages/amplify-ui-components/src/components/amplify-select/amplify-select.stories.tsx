import { storiesOf } from '@storybook/html';

const selectStories = storiesOf('amplify-select', module);

selectStories.add('default with country dial codes', () => {
  return `<amplify-select></amplify-select>`;
});
