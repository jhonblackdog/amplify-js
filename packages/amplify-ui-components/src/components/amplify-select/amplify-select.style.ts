import { css } from 'emotion';

export const select = css`
  display: flex;
  padding: 14px;
  font-size: 14px;
  color: var(--deep-squid-ink);
  background-color: #fff;
  background-image: none;
  border: 1px solid var(--light-grey);
  border-right: none;
  border-radius: 3px 0 0 3px;
  box-sizing: border-box;
  margin: 10px 0 10px 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  flex-basis: auto;
  width: 7%;
  flex: 1;

  background-image:
    linear-gradient(45deg, transparent 50%, gray 50%),
    linear-gradient(135deg, gray 50%, transparent 50%),
    linear-gradient(to right, #ccc, #ccc);
  background-position:
    calc(100% - 20px) calc(1em + 8px),
    calc(100% - 15px) calc(1em + 8px),
    calc(100% - 2.5em) 0.5em;
  background-size:
    6px 5px,
    6px 5px,
    0px 1.5em;
  background-repeat: no-repeat;
`;