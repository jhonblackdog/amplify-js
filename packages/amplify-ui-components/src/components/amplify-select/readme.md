# amplify-select



<!-- Auto Generated Below -->


## Properties

| Property        | Attribute        | Description                                                                                                                 | Type                                             | Default            |
| --------------- | ---------------- | --------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------ | ------------------ |
| `fieldId`       | `field-id`       | Used for id field                                                                                                           | `string`                                         | `undefined`        |
| `options`       | --               | The options of the select input. Must be an Array of Objects with an Object shape of {label: string, value: string\|number} | `SelectOptions<number> \| SelectOptions<string>` | `countryDialCodes` |
| `styleOverride` | `style-override` | (Optional) Overrides default styling                                                                                        | `boolean`                                        | `false`            |


## Dependencies

### Used by

 - [amplify-examples](../amplify-examples)

### Graph
```mermaid
graph TD;
  amplify-examples --> amplify-select
  style amplify-select fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
