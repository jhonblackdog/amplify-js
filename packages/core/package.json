{
  "name": "@aws-amplify/core",
  "version": "1.0.26-unstable.1",
  "description": "Core category of aws-amplify",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/src/index.js",
  "typings": "./dist/esm/src/index.d.ts",
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "test": "tslint 'src/**/*.ts' && jest --coverage",
    "build-with-test": "npm test && npm run build",
    "build:cjs": "tsc -p tsconfig.es5.json && rollup -c",
    "build:esm": "tsc -p tsconfig.es2015.json",
    "build": "npm run clean && npm run build:cjs && npm run build:esm",
    "clean": "rimraf tsc-out dist",
    "format": "tsfmt --useTsfmt tsfmt.json -r src/**/*.ts",
    "lint": "tslint 'src/**/*.ts'"
  },
  "react-native": {
    "./lib/ClientDevice": "./lib/ClientDevice/reactnative.js",
    "./lib/RNComponents": "./lib/RNComponents/reactnative.js",
    "./lib/StorageHelper": "./lib/StorageHelper/reactnative.js"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/aws/aws-amplify.git"
  },
  "author": "Amazon Web Services",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/aws/aws-amplify/issues"
  },
  "homepage": "https://github.com/aws/aws-amplify#readme",
  "devDependencies": {
    "find": "^0.2.7",
    "prepend-file": "^1.3.1",
    "prettier": "^1.7.4",
    "typescript": "^3.2.2"
  },
  "dependencies": {
    "aws-sdk": "2.329.0",
    "url": "^0.11.0"
  },
  "jest": {
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": "<rootDir>../../node_modules/ts-jest/preprocessor.js"
    },
    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(tsx?|jsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "json",
      "jsx"
    ],
    "testEnvironment": "jsdom",
    "mapCoverage": true,
    "coverageThreshold": {
      "global": {
        "branches": 0,
        "functions": 0,
        "lines": 0,
        "statements": 0
      }
    },
    "testURL": "http://localhost/",
    "coveragePathIgnorePatterns": [
      "/node_modules/"
    ]
  }
}
