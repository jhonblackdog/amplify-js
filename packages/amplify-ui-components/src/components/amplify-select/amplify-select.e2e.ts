import { newE2EPage } from '@stencil/core/testing';

describe('Select Component:', () => {
  it('should render a amplify-select', async () => {
    const page = await newE2EPage();
    await page.setContent(`<amplify-select></amplify-select>`);
    const el = await page.find('amplify-select');
    expect(el).not.toBeNull();
  });
});