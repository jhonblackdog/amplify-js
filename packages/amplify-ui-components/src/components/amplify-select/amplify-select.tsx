import { Component, Prop, h } from '@stencil/core';
import { select } from './amplify-select.style';
import { styleNuker } from '../../common/helpers';
import { AMPLIFY_UI_SELECT } from '../../common/constants';
import { SelectOptions } from './amplify-select-interface';
import countryDialCodes from '../../common/country-dial-codes';

@Component({
  tag: 'amplify-select',
})
export class AmplifySelect {
  /** (Optional) Overrides default styling */
  @Prop() styleOverride: boolean = false;
  /** Must be an Array of Objects with an Object shape of {label: string, value: string|number} */
  @Prop() options?: SelectOptions;

  splitIntoSeparateOptionsForSelect(opts: SelectOptions) {
    let content = [];
    opts.forEach(opt => content.push(<option value={opt.value}>{opt.label}</option>));
  
    return content;
  };

  render() {
    return (
      <select class={styleNuker(this.styleOverride, AMPLIFY_UI_SELECT, select)}>
        {this.options ? this.splitIntoSeparateOptionsForSelect(this.options): this.splitIntoSeparateOptionsForSelect(countryDialCodes)}
      </select>
    );
  }
}