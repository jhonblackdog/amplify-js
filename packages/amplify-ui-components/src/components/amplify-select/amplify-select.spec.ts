import { newSpecPage } from '@stencil/core/testing';
import { AmplifySelect } from './amplify-select';

describe('amplify-select: ', () => {
  describe('Custom logic cases ->', () => {
    describe('Class logic ->', () => {
      let select; 

      beforeEach(() => {
        select = new AmplifySelect();
      });

      it('should have style override set to true when style-override is passed as a prop from parent', () => {
        select.styleOverride = true;
        expect(select.styleOverride).toBe(true);
      });

      it('should use custom options when passed from parent', () => {
        const options = [
          {label: 'Gogi', value: '1'},
          {label: 'Gucci', value: '2'},
          {label: 'Foooci', value: '3'}
        ];

        select.options = options;

        expect(select.options).toBeDefined();
        expect(select.options).toEqual(options);
      });

      it('should throw an error if label from options is not a string', () => {
        const options = [{ label: 1, value: 1 }];

        select.options = options;

        expect(select.options).toBeDefined();
        expect(select.splitIntoSeparateOptionsForSelect).toThrowError();
      });
    });
  });
  describe('Default cases ->', () => {
    describe('Class logic ->', () => {
      let select;

      beforeEach(() => {
        select = new AmplifySelect();
      });

      it('should have style override set to false', () => {
         expect(select.styleOverride).toBe(false);
      });
      it('should have options be defined ', () => {    
        expect(select.options).toBeDefined();
      });
    });
    describe('HTML logic ->', () => {
      it('should render with all country code dials as default', async () => {
        const page = await newSpecPage({
          components: [AmplifySelect],
          html: `<amplify-select></amplify-select>`
        });
    
        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
