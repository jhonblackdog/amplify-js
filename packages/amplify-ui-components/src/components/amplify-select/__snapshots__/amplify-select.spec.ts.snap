// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-select:  Default cases -> HTML logic -> should render an empty \`placeholder\` label and value of \`1\` 1`] = `
<amplify-select>
  <select class="amplify-ui--amplify-select css-10dovbt">
    <option value="1"></option>
  </select>
</amplify-select>
`;
